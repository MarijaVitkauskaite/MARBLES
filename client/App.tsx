import 'react-native-gesture-handler';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { useState, createContext } from 'react';
import Login from './views/login/login';
import Calendar from './views/calendar/calendar';
import Habits from './views/habits/habits';
import Register from './views/register/register';
import AddHabit from './components/AddHabit/AddHabit';
import { User } from '../lib/api-intefaces';
import { userContext } from './user-context';

// TODO  a separate router component file && check
const Stack = createStackNavigator();
function MyStack() {

  return (
    <Stack.Navigator screenOptions={{ headerShown: false }}>
      <Stack.Screen name="Login" component={Login} />
      <Stack.Screen name="Register" component={Register} />
      <Stack.Screen name="Habits" component={Habits} />
      <Stack.Screen name="AddHabit" component={AddHabit} />
      <Stack.Screen name="Calendar" component={Calendar} />
    </Stack.Navigator>

  );
}


export default function App() {

  const user = useState<User>({ userId: "", email: "", habits: [] });
  return (
    <NavigationContainer>
      <userContext.Provider value={user} >
        <MyStack />
      </userContext.Provider>
    </NavigationContainer>
  );
}
