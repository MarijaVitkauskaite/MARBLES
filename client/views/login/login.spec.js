import React from 'react';
// import renderer, { act } from 'react-test-renderer';
//All for old version. Needs update.
import Login from './login';
import { Alert } from 'react-native';
import { render, act, fireEvent, screen } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import App from '../../App'


jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.spyOn(Alert, 'alert');

// const element = screen.getByTestId('unique-id');

describe('Login page', () => {
  test('should render correctly', () => {
    const tree = render(<Login />)
    const treejson = tree.toJSON();
    expect(treejson).toMatchSnapshot();
  });
  test('should trigger alert when click login button without email input', async () => {

    render(<Login />)
    // const mockOnPress = jest.fn()
    const button = screen.getByTestId("login-button");
    fireEvent.press(button)
    const errorMessageText = 'Please enter email address'
    // await act(async () => button.onPress());
    expect(Alert.alert).toHaveBeenCalledWith(errorMessageText)

  });

  test('should trigger alert when click login button with email input but without password input', async () => {
    render(<Login />)
    const button = screen.getByTestId("login-button");
    const emailInput = screen.getByTestId("email-input");
    fireEvent.changeText(emailInput, 'test@gmail.com');
    fireEvent.press(button)
    const errorMessageText = 'Please enter password'
    expect(Alert.alert).toHaveBeenCalledWith(errorMessageText)

  });

  test('should trigger alert when passing wrong email-password pairs', async () => {
    render(<Login />)
    const button = screen.getByTestId("login-button");
    const emailInput = screen.getByTestId("email-input");
    const passwordInput = screen.getByTestId("password-input");
    fireEvent.changeText(emailInput, 'test@gmail.com');
    fireEvent.changeText(passwordInput, 'testpassword');
    fireEvent.press(button)
    const errorMessageText = 'Please register'
    expect(Alert.alert).toHaveBeenCalledWith(errorMessageText)
  });

  test('should go to register page when click the button', async () => {
    const component = (
      <NavigationContainer>
        <App />
      </NavigationContainer>
    );
    render(component);
    const button = screen.getByTestId("register-button");
    fireEvent.press(button)
    const registerEmailInput = screen.getByTestId("email-input-register");
    expect(registerEmailInput).toBeTruthy();
  })
})

// it ('does stuff', () => {
//   const mock = jest.fn()
//   const component = render(<Search onSearchTextChange={mock}/>)
//   fireEvent.changeText(component.findByType(TextInput), 'test')
//   expect(mock).toHaveBeenCalledWith('test')
// })


// Alert.alert('Title', 'Message', [{text: 'OK', onPress: mockOnPress}])

// /**
//  * here we inspect the mock to get:
//  * - the latest call to Alert.alert (`calls[0]`)
//  * - its third argument (`calls[0][2]`), which is the buttons config array
//  * - the first button in the array and its onPress (`calls[0][2][0].onPress()`)
//  */
// Alert.alert.mock.calls[0][2][0].onPress()

// expect(mockOnPress).toBeCalled()