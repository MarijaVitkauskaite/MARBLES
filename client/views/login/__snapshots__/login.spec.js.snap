// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page should render correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7197AC",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/Frame.png",
      }
    }
    style={
      Object {
        "bottom": "70%",
        "position": "absolute",
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../assets/Marbles.png",
      }
    }
    style={
      Object {
        "bottom": "47.5%",
        "height": 300,
        "position": "absolute",
        "width": 300,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#D9D9D9",
        "borderRadius": 30,
        "bottom": "-20%",
        "height": 50,
        "marginBottom": 20,
        "opacity": 0.6,
        "width": "70%",
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="EMAIL"
      placeholderTextColor="#353535"
      style={
        Object {
          "alignItems": "stretch",
          "flex": 1,
        }
      }
      testID="email-input"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#D9D9D9",
        "borderRadius": 30,
        "bottom": "-20%",
        "height": 50,
        "marginBottom": 20,
        "opacity": 0.6,
        "width": "70%",
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="PASSWORD"
      placeholderTextColor="#353535"
      secureTextEntry={true}
      style={
        Object {
          "alignItems": "stretch",
          "flex": 1,
        }
      }
      testID="password-input"
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": "-20%",
        "opacity": 1,
      }
    }
    testID="login-button"
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/Group.png",
        }
      }
      style={
        Object {
          "height": 50,
          "width": 100,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": "-20%",
        "opacity": 1,
      }
    }
    testID="register-button"
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/Register.png",
        }
      }
      style={
        Object {
          "top": 120,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
